### PR TITLE
[kube-prometheus-stack ]Upgrade to 43.0.0

### LINKED ISSUE #2788: [kube-prometheus-stack] Grafana not deployed after upgrading to 42.2.0
### Describe the bug a clear and concise description of what the bug is.

I had the chart deployed at version 42.0.2, with the following `grafana.persistence` section:
```
grafana:
  persistence:
    type: statefulset
    enabled: true
    storageClassName: rook-ceph-block
    accessModes:
      - ReadWriteOnce
    size: 10Gi
```

Before [this Grafana commit](https://github.com/grafana/helm-charts/commit/45f71da2c9055f6de23761434a8fc2472559ca53)  this was enough to get Grafana deployed with a StatefulSet.

After upgrading to chart version 42.2.0 Grafana StatefulSet disappeared.

This is because version 42.2.0 bumped Grafana chart to 6.45.0 which introduced the mentioned commit.

I didn't find this documented anywhere, the chart major version is still 42 and there's no way users can take the required actions to prepare for this upgrade.

I think this should be documented with a major version change.

### What's your helm version?

version.BuildInfo{Version:"v3.10.2", GitCommit:"50f003e5ee8704ec937a756c646870227d7c8b58", GitTreeState:"clean", GoVersion:"go1.18.8"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:57:26Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"linux/amd64"} Kustomize Version: v4.5.7 Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.4+k3s1", GitCommit:"c3f830e9b9ed8a4d9d0e2aa663b4591b923a296e", GitTreeState:"clean", BuildDate:"2022-08-25T03:45:26Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

kube-prometheus-stack

### What's the chart version?

42.2.0

### What happened?

Grafana Pod was not deployed

### What you expected to happen?

Grafana Pod should be deployed

### How to reproduce it?

Deploy version 42.2.0 with:
```
grafana:
  persistence:
    type: statefulset
    enabled: true
    storageClassName: rook-ceph-block
    accessModes:
      - ReadWriteOnce
    size: 10Gi
```

### Enter the changed values of values.yaml?

```
grafana:
  persistence:
    type: statefulset
    enabled: true
    storageClassName: rook-ceph-block
    accessModes:
      - ReadWriteOnce
    size: 10Gi
```

### Enter the command that you execute and failing/misfunctioning.

helm install --release-name kube-prometheus-stack --namespace monitoring --version 42.2.0 -f values.yaml --include-crds prometheus-community/kube-prometheus-stack

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by irizzant**
Signed-off-by: irizzant <i.rizzante@gmail.com>

#### What this PR does / why we need it
This PR upgrades the chart to version 43.0.0 in order to document a change to Grafana sub chart which prevents Grafana Pod to be deployed.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #2788

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Can you please also add the update to prom operator  0.61.1 and also update the docker image versions?

**Comment by irizzant** [comment]
@monotek I don't know how to make the branch writable but I think I copied everything you asked me.

File: charts/kube-prometheus-stack/README.md [hunk 1]
```diff
@@ -80,6 +80,51 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### From 42.x to 43.x
+This version upgrades Prometheus-Operator to v0.61.1, Prometheus to v2.40.5 and Thanos to v0.29.10
```
Comment (monotek) [inline charts/kube-prometheus-stack/README.md:84]:
```suggestion
This version upgrades Prometheus-Operator to v0.61.1, Prometheus to v2.40.5 and Thanos to v0.29.0
```

**Review by monotek — CHANGES_REQUESTED** [review]
all the changes to the crds, dashboards and so on are missing.

about write access: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by irizzant** [comment]
@monotek I've also added the missing resources you requested

**Review by monotek — CHANGES_REQUESTED** [review]
I just checked the update from 41.0.0 to 42.0.0 to 42.2.0 and did not hit any Grafana related errror using the default configuration of our chart, so i dont think we have to extend our upgrading docs at all.

So i would remove the grafana instructions here as it might be misleading for users which are just using the default config.

**Comment by irizzant** [comment]
@monotek apparently they released a new Grafana chart 6.45.1 which contains https://github.com/grafana/helm-charts/pull/2041 and allows the previous configuration to work again.

If you prefer I can upgrade the dependency instead

**Comment by monotek** [comment]
Yes, please update dependency

**Comment by cod-r** [comment]
prometheus-operator v0.61.1 added a new spec field `spec.alertmanagerConfigMatcherStrategy.type:`. See: https://github.com/prometheus-operator/prometheus-operator/issues/3737#issuecomment-1326667523
I can't find this option in kube-prometheus-stack `values.yaml` file

**Review by monotek — APPROVED** [review]

**Comment by cod-r** [comment]
Hi @monotek, you merged the PR but I still can't see the option to enable `spec.alertmanagerConfigMatcherStrategy.type` in `values.yaml`

It should be:
```
alertmanager:
  alertmanagerSpec:
    alertmanagerConfigMatcherStrategy:
      type: None
```

Am I missing something?

**Pull request merged**